src/sources/http_scrape/scrape.rs

```diff
@@ -0,0 +1,243 @@
+//! Generalized HTTP scrape source.
+//! Scrapes an endpoint at an interval, decoding the HTTP responses into events.
+
+use bytes::{Bytes, BytesMut};
+use chrono::Utc;
+use futures_util::FutureExt;
+use http::{response::Parts, Uri};
+use snafu::ResultExt;
+use std::collections::HashMap;
+use tokio_util::codec::Decoder as _;
+
+use crate::{
+    codecs::{Decoder, DecodingConfig},
+    config::{SourceConfig, SourceContext, SourceDescription},
+    http::Auth,
+    serde::default_decoding,
+    serde::default_framing_message_based,
+    sources,
+    tls::{TlsConfig, TlsSettings},
+    Result,
+};
+use codecs::{
+    decoding::{DeserializerConfig, FramingConfig},
+    StreamDecodingError,
+};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{log_schema, LogNamespace, Output},
+    event::Event,
+};
+
+/// The name of this source
+pub(crate) const NAME: &str = "http_scrape";
+
+/// Configuration for the `http_scrape` source.
+#[configurable_component(source)]
+#[derive(Clone, Debug)]
+pub struct HttpScrapeConfig {
```

Comment:
I'd suggest another follow-up issue for compression support.

Reply:
Opened #13888

, and linked it and the OAuth one in the initial comment.